src/hotspot/share/gc/shared/c2/barrierSetC2.hpp

```diff
@@ -102,22 +102,22 @@ class C2AccessValuePtr: public C2AccessValue {
 class C2Access: public StackObj {
 protected:
   DecoratorSet      _decorators;
-  BasicType         _type;
   Node*             _base;
   C2AccessValuePtr& _addr;
   Node*             _raw_access;
+  BasicType         _type;
   uint8_t           _barrier_data;
 
   void fixup_decorators();
 
 public:
   C2Access(DecoratorSet decorators,
-           BasicType type, Node* base, C2AccessValuePtr& addr) :
+           Node* base, C2AccessValuePtr& addr, BasicType type) :
```

Comment:
I think it would be cleaner to leave the argument order alone here, and only change the field order. This would guarantee we do not change anything in APIs, which simplifies future changes and backports.